diff -urN a/meson.build b/meson.build
--- a/meson.build	1969-12-31 19:00:00.000000000 -0500
+++ b/meson.build	2023-01-06 00:33:02.698341442 -0500
@@ -0,0 +1,8 @@
+project(
+  'winpthreads', 'c',
+  default_options : [
+    'buildtype=debugoptimized',
+  ]
+)
+
+subdir('mingw-w64-libraries')
diff -urN a/mingw-w64-libraries/meson.build b/mingw-w64-libraries/meson.build
--- a/mingw-w64-libraries/meson.build	1969-12-31 19:00:00.000000000 -0500
+++ b/mingw-w64-libraries/meson.build	2023-01-06 00:33:02.701341454 -0500
@@ -0,0 +1 @@
+subdir('winpthreads')
diff -urN a/mingw-w64-libraries/winpthreads/meson.build b/mingw-w64-libraries/winpthreads/meson.build
--- a/mingw-w64-libraries/winpthreads/meson.build	1969-12-31 19:00:00.000000000 -0500
+++ b/mingw-w64-libraries/winpthreads/meson.build	2023-01-06 00:33:02.704341466 -0500
@@ -0,0 +1,37 @@
+rc = import('windows').compile_resources(
+  'src/version.rc',
+)
+
+winpthread_sources = [
+  'src/barrier.c',
+  'src/clock.c',
+  'src/cond.c',
+  'src/misc.c',
+  'src/mutex.c',
+  'src/nanosleep.c',
+  'src/ref.c',
+  'src/rwlock.c',
+  'src/sched.c',
+  'src/sem.c',
+  'src/spinlock.c',
+  'src/thread.c',
+  rc,
+]
+
+shared_library(
+  'winpthread',
+  winpthread_sources,
+  c_args : [
+    '-DWIN32_LEAN_AND_MEAN',
+    '-DIN_WINPTHREAD',
+    '-DWINPTHREAD_DBG=1',
+    '-D__USE_MINGW_ANSI_STDIO=0',
+    '-no-pthread',
+  ],
+  link_args : [
+    '-Wl,--no-undefined',
+  ],
+  include_directories : 'include',
+  install : true,
+  version: '1.0.0',
+)
